macros/sql/get_tables_by_prefix_sql.sql

```diff
@@ -1,25 +1,25 @@
-{% macro get_tables_by_prefix_sql(schema, prefix, exclude='') %}
+{% macro get_tables_by_prefix_sql(schema, prefix, exclude='', database=target.database) %}
     {{ adapter_macro('dbt_utils.get_tables_by_prefix_sql', schema, prefix, exclude) }}
 {% endmacro %}
 
-{% macro default__get_tables_by_prefix_sql(schema, prefix, exclude='') %}
+{% macro default__get_tables_by_prefix_sql(schema, prefix, exclude='', database=target.database) %}
 
         select distinct 
-            table_schema || '.' || table_name as ref
-        from information_schema.tables
+            table_schema as "table_schema", table_name as "table_name"
+        from {{database}}.information_schema.tables
```

Comment:
Fully agreed. I had tried using the `adapter.quote()` method (necessary in the BQ version below, given our BQ test project name), but it caused errors on Snowflake. Noted for our future selves.

Reply:
We spent _a lot_ of time getting this working right with quoting in dbt-core. I think this code is fine for now, but in general, we'll probably want to create some sort of object (analogous to Relations) which encapsulates a database + schema and any applicable quoting configs.

This code won't work if a database is named `my-database` on eg. Snowflake. While a user could pass in their database string as `"my-database"`, that would cause problems when we create the Relation object in `get_tables_by_prefix`. I'm ok with shipping this as-is, but let's keep examples like this in mind as we improve dbt's understanding of databases and quoting